### PR TITLE
use libav to play SRT streams

### PR DESCRIPTION
**Pull request by erdnaxe**
Secure Reliable Tranport (SRT) is an open source low-latency streaming protocol. It is supported by ffmpeg in Debian Bullseye.

It would be nice to be able to play SRT streams in MPV. As MPV can use libav, this seems to be quite a simple contribution.

### CODE REVIEW

**Comment by Akemi** [comment]
should probably be added to https://github.com/mpv-player/mpv/blob/master/TOOLS/osxbundle/mpv.app/Contents/Info.plist#L203-L221 too? does the exe manifest for windows have something similar?

**Comment by erdnaxe** [comment]
Screenshot from 2020-10-03 21-42-40

lavf successfully discovers the mpegts format, but is unable to read frames as shown on the screenshot.
I use [srt-live-transmit](https://github.com/Haivision/srt/blob/master/apps/srt-live-transmit.cpp) with `srt-live-transmit srt://127.0.0.1:5001?mode=listener srt://127.0.0.1:5002?mode=listener`, OBS Studio streaming on `srt://127.0.0.1:5001?streamid=demo`, then `mpv srt://127.0.0.1:5002?streamid=demo`.

**Comment by erdnaxe** [comment]
Screenshot from 2020-10-04 11-11-52

My OBS options were wrong. I forced MPEGTS and now it is fully working \o/

File: stream/stream_lavf.c [hunk 1]
```diff
@@ -310,6 +310,10 @@ static int open_f(stream_t *stream)
         av_dict_set(&dict, "timeout", "0", 0);
     }
 
+    if (strncmp(filename, "srt", 3) == 0) {
+        stream->demuxer = "lavf";
```
Comment (jeeb) [inline stream/stream_lavf.c:314]: Are you sure this is required? I have mostly been poking with pure UDP and it moved nicely through probing to MPEG-TS for example.

I'd rather not add more kludges if it's possible :) .

**Review by jeeb — COMMENTED** [review]
Thank you for your interest.

Otherwise I think this seems quite alright, but I'm just not sure if that demuxer force needs to be there?

File: stream/stream_lavf.c (hunk 1 above)
Comment (erdnaxe) [reply stream/stream_lavf.c]: You are right, without the demuxer forced, MPV is still able to read my stream :)

If you want to merge something more clean I can close this MR and recreate a new branch with this modification.

**Comment by Akemi** [comment]
you would need to squash these changes and the commit message should follow our contribution guidelines https://github.com/mpv-player/mpv/blob/master/DOCS/contribute.md

**Comment by erdnaxe** [comment]
This squash should respect the contribution guidelines. If I missed anything please let me know and I will gladly fix it.

**Comment by jeeb** [comment]
@erdnaxe , I don't see any other issues here. Would it be OK to reword the first line(s) to something like
```
stream_lavf: enable SRT protocol support through FFmpeg

Additionally, announce support for the protocol in Mac and Linux
application metadata.
```
?

**Comment by Akemi** [comment]
the commit should be prefixed with "stream_lavf:" or "stream:", analog to the other commits https://github.com/mpv-player/mpv/commits/f5ee3051e42ac899bdd41cb0a3c53c0071d96128/stream/stream_lavf.c

**Comment by jeeb** [comment]
Oh, and additionally almost forgot: A description would be needed in https://github.com/mpv-player/mpv/blob/master/DOCS/man/mpv.rst . See how the `smb://` entry as an example.

**Comment by jeeb** [comment]
Ahh, actually no. Now I finally read the first entry :P .

Most FFmpeg-passed protocols are specified by the initial one.

**Review by jeeb — APPROVED** [review]
Thank you. LGTM.

**Pull request merged**